fixing multi-depends, starting on fetch precip data

wdwatkins opened the pull request:
I think this is working, it started a gdp job at least.  We'll find out in the morning...

wdwatkins committed:
fixing multi-depends, starting on fetch precip data

jordansread commented on scripts/fetch/fetchPrecip.R:
```diff
@@ -37,9 +36,9 @@ fetch.precip <- function(viz){
   attr(startDate, 'tzone') <- "UTC"
   
-  states <- viz[['states']]
+  counties <- readDepends(viz)[['counties']] #TODO: need to get counties, send into getPrecip
```
still foggy on the depends naming. this works when you keep the depends in an array:
```r
counties <- readDepends(viz)[['counties']]
```

jordansread replied:
so why do we need to name?

lindsayplatt replied:
just talked about this in person, but for future reference -- only time we need a name is when we have a generic function that gets used with multiple depends. E.g. a processor step uses `dailyData` as input but we use the processor step two different times with different data; we would need something generic so the processor function could use whichever data actually came in.

jordansread commented on viz.yaml:
```diff
@@ -86,20 +84,36 @@ fetch:
+    mimetype: text/csv
+    refetch: true
+    depends:
+      counties: counties
```
The PR looks good, but I don't follow this `thing1:thing1`, since the depends come back as a named list out of `readDepends()`

lindsayplatt commented on viz.yaml:
```diff
@@ -86,20 +84,36 @@ fetch:
     visualizer: hurricane_map
-    depends: ["base-map"]#, "precip-category", "precip-colors", "discharge-sparks"]
+    depends: 
+      base-map: "base-map"#, "precip-category", "precip-colors", "discharge-sparks"]
```
watch out - you're missing the first `[`

lindsayplatt replied:
oh wait - I see now, it's all commented out so doesn't matter

wdwatkins committed:
merge conflict

Merge branch 'master' of https://github.com/USGS-VIZLAB/hurricane-irma

# Conflicts:
#	viz.yaml
